### PR TITLE
Theme Totals

### PR DESCRIPTION
**Pull request by DanaJomar**

### CODE REVIEW

**Comment by LyzandeR** [comment]
Happy new year both! 

The PR looks great. We need to update the vignettes now that we have more themes and before we release the new version. 

@clemens-zauchner  do you think we should make a themes vignette? I am thinking in favour of it.

**Comment by clemens-zauchner** [comment]
I think so too, PR looks great and I really like the themes. I have added two comments, minor changes really. 
@LyzandeR vignettes should be definitely there, makes it easier to use the themes.

**Comment by LyzandeR** [comment]
Can I suggest something? Why don't we add `...` to `add_theme` and then add to the details of the documentation the extra arguments for `add_theme_totals`. I think having more than one function for the themes will make it difficult for the user to find them.

**Comment by LyzandeR** [comment]
So, after we merge, I ll remove the themes from the main tutorial vignette and add a new themes vignette.

File: R/add_theme_totals.R [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+#' \code{add_theme_totals} will add an Excel-like theme to tableHTML 
+#'  and hightlights one or more total-rows.
+#'
+#' \code{add_theme_totals} will add an Excel-like theme to tableHTML.
+#' Column widths are not provided with the theme.
+#' Please use the width argument for column widths.
+#'
+#' @param tableHTML A tableHTML object.
+#' 
+#' @param color A character vector to specify the desired color. It can contain
```
Comment (LyzandeR) [inline R/add_theme_totals.R]: Here in the documentation, can we add what is the difference between one or two colours?

**Comment by DanaJomar** [comment]
I pushed the changes, please take a look again.
Should I add  `... ` to `add_theme` ? or is it still not decided? --- I like the idea though.

And @LyzandeR Happy new year to you too 😃

**Comment by LyzandeR** [comment]
@DanaJomar Then if you agree as well let's do it. My rational is that the user should only need to know one function for the themes. Now, if they want to investigate further, they can always use the functions we created directly. So, I would actually include the arguments `total_rows` and `id_column` in `add_theme` and would include in the documentation that these arguments are only used when the theme is `totals` (since we have all the arguments within `add_theme`, there is no point of having `totals-green` and `totals-blue` since the user can use the argument color).

Also, please add a couple of examples in the `add_theme` examples.

**Comment by clemens-zauchner** [comment]
As long as the totals theme has a default colour I think it is fine, so `add_theme` would still be generic and work with only the `theme` argument. I think that's important for simplicity.

**Comment by LyzandeR** [comment]
Also, should the argument be `color` or `colour`? I see that we use the word `colour` throughout the documentations of the other functions. Thoughts?

**Comment by clemens-zauchner** [comment]
Not necessarily confuse them, the reason we decided to keep the `add_theme` generic was to offer themes that work without any additional customisation. As soon as we start adding arguments, there will be the need to customise themes, which is in a way not what was intended in the first place. 
I still think that the themes should be generic without the need to customise, however there is a benefit in making customisation possible, albeit optional. The question is: how can we make sure both can be achieved.

**Comment by clemens-zauchner** [comment]
The spelling of `colour` is my bad, I have messed it up when introducing conditional formatting as I am using both. ggplot2 allows both and that might be a way to make sure changes won't be breaking ones.

**Comment by LyzandeR** [comment]
Cool. The only way then is to just mention and link-refer to the underlying functions in the documentation of `add_theme`. I don't see any other way. Doesn't it feel weird if you say in the documentation of `add_theme` something like "if you want to change the colour, please use `add_theme_totals`"? Because that would be the only way to offer the customisation.

**Comment by clemens-zauchner** [comment]
I think using `...` for additional arguments is fine, as long as the default works. So if I wanted to use the totals theme it would work, if I want to customise it, I can provide additional arguments.

**Comment by LyzandeR** [comment]
Then we would need to use `color` though, otherwise the `...` would only be used for the `id_column`, right? Which creates a need to document it :)

**Comment by LyzandeR** [comment]
Also, I think we should add both `color` and `colour` then.

**Comment by LyzandeR** [comment]
I like the use of `...` and the documentation that you added in the comment. How would the color work though? If the user would leave it blank, what would it default to? Would you keep `totals-blue` and `totals-green` or would you just have `totals`?

**Comment by DanaJomar** [comment]
I would chose a default color like ``steelblue`` or ``darkreen`` (currently the default is `purple4` but I don't like it anymore) and use just ``"totals"``

**Comment by DanaJomar** [comment]
> Also, I think we should add both `color` and `colour` then.

Do we then put both in each function as well as in  the documentations? or is there a more efficient 'standard' way to use two names for the same parameter?

**Comment by LyzandeR** [comment]
> I would choose a default color like `steelblue` or `darkgreen` (currently the default is `purple4` but I don't like it anymore) and use just `"totals"`

Cool, let's do this then.

**Comment by clemens-zauchner** [comment]
This is how ggplot2 does it for functions: https://github.com/tidyverse/ggplot2/blob/master/R/zxx.r

And here you can see how they do it for arguments:
https://github.com/tidyverse/ggplot2/blob/master/R/aes.r

**Comment by LyzandeR** [comment]
> > Also, I think we should add both `color` and `colour` then.
> 
> Do we then put both in each function as well as in the documentations? or is there a more efficient 'standard' way to use two names for the same parameter?

@clemens-zauchner  has used `color` in one function and `colour` in another function. It's best if we stick to one. I would go for `colour` because we use it more, unless anyone objects.

**Comment by DanaJomar** [comment]
So now the `...` are in, I will take a look at the links in @clemens-zauchner comment, and see how much is this 'bit' of coding.
Small note though, you both probably know already, but I think it is worth mentioning that `css` does not take `colour` as a style definition (only `color`),  which means if we want to go for ``colour`` only, then  NO FAST REPLACE, a thing I had to learn the hard way.

**Comment by LyzandeR** [comment]
> Correct me if I'm wrong but it seems that passing the aesthetics between layers is the main issue in `ggplot2::aes` , the whole thing looks a bit like an overkill for our purpose, the function `standardise_aes_names` is our main goal here right?
> 
>   So what i was thinking is: we can pass the customisation arguments as a named list like this (`cstm = list(color = 'darkgreen', total_rows = NULL, id_column = FALSE)`), then we can standardise the names with `gsub` like it was done with `standardise_aes_names` , and then the rest pretty much stays the same.  
> 
> @LyzandeR @clemens-zauchner Thoughts?

I feel that `color` is probably simpler. Otherwise, you need a list and it becomes a bit difficult to use without formal arguments.

**Pull request merged**